File: Tattvamasi/app/src/main/java/com/ayyappasamaaj/tattvamasi/model/ParentListItem.java
package com.ayyappasamaaj.tattvamasi.model;

public class ParentListItem {
    private String name;

    public ParentListItem() {}

    public ParentListItem(String name) {
        this.name = toTitleCase(name);
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    private static String toTitleCase(String givenString) {
        String[] arr = givenString.split(" ");
        StringBuffer sb = new StringBuffer();

        for (int i = 0; i < arr.length; i++) {
            sb.append(Character.toUpperCase(arr[i].charAt(0)))
                    .append(arr[i].substring(1)).append(" ");
        }
        return sb.toString().trim();
    }
}

File: Tattvamasi/app/src/main/java/com/ayyappasamaaj/tattvamasi/model/ParentListItem.kt
package com.ayyappasamaaj.tattvamasi.model

class ParentListItem(name: String) {
    var name: String? = null

    init {
        this.name = toTitleCase(name)
    }

    companion object {
        private fun toTitleCase(givenString: String): String {
            val arr = givenString.split(" ").toTypedArray()
            val sb = StringBuffer()
            for (i in arr.indices) {
                sb.append(Character.toUpperCase(arr[i][0]))
                    .append(arr[i].substring(1)).append(" ")
            }
            return sb.toString().trim { it <= ' ' }
        }
    }
}